### PR TITLE
Implement Full Flow-Typing of Some Scenes

### PR DESCRIPTION
**Pull request by kylanhurt**
**This commit only covers scenes alphabetically from `/AddToken` to `/Request`** There is some tangential work done in other folders, but  a lot of that was only done in order to as a matter of convenience.

Related Asana Task: https://app.asana.com/0/361770107085503/511473289504673/f #

### CODE REVIEW

File: src/modules/UI/scenes/CryptoExchange/CryptoExchangeConfirmTransactionModalComponent.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ import s from '../../../../locales/strings.js'
 import {GuiWallet} from '../../../../types'
 import THEME from '../../../../theme/variables/airbitz'
 
-type Props = {
+type CryptoExchangeConfirmTransactionModalOwnProps = {
   style: any,
```
Comment (wolverineks) [inline src/modules/UI/scenes/CryptoExchange/CryptoExchangeConfirmTransactionModalComponent.js]: Please avoid using any
At the bare minimum, styles should be typed `Object` or `{}`

Comment (wolverineks) [reply src/modules/UI/scenes/CryptoExchange/CryptoExchangeConfirmTransactionModalComponent.js]: If you see `any` when adding types, please replace them with more precise types

File: src/modules/UI/scenes/EdgeLogin/EdgeLoginSceneComponent.js [hunk 2]
```diff
@@ -5,15 +5,15 @@ import { PrimaryButton, SecondaryButton } from '../../components/Buttons/index'
 import s from '../../../../locales/strings.js'
 import type { AbcLobby } from 'airbitz-core-types'
 
-type Props = {
+type EdgeLoginSceneProps = {
   style: any,
```
Comment (wolverineks) [inline src/modules/UI/scenes/EdgeLogin/EdgeLoginSceneComponent.js]: Please replace `any` types if you come across them when adding types

File: src/modules/UI/scenes/EditToken/EditToken.ui.js [hunk 3]
```diff
@@ -14,22 +14,38 @@ import styles from './style.js'
...
+import type {CustomTokenInfo} from '../../../../types.js'
+
 import _ from 'lodash'
 
-export type DispatchProps = {
+export type EditTokenDispatchProps = {
   showDeleteTokenModal: () => void,
   hideDeleteTokenModal: () => void,
   deleteCustomToken: (string, string) => void,
   editCustomToken: (string, string, string, string, string, string) => void
```
Comment (wolverineks) [inline src/modules/UI/scenes/EditToken/EditToken.ui.js]: If you come across functions that have unnamed parameters in their type signature, please add names
`(string, string, string, string) => {...}` ---> `(walletId: string, walletName: string, balance: string) => {...}`

File: src/modules/UI/scenes/EditToken/EditToken.ui.js [hunk 4]
```diff
@@ -14,22 +14,38 @@ import styles from './style.js'
...
+  customTokens: Array<CustomTokenInfo>,
+  deleteTokenModalVisible: boolean,
+  editCustomTokenProcessing: boolean,
+  deleteCustomTokenProcessing: boolean
+}
+
+export type EditTokenOwnProps = {
+  walletId: string,
+  addTokenPending: Function,
+  addToken: Function,
```
Comment (wolverineks) [inline src/modules/UI/scenes/EditToken/EditToken.ui.js]: Please avoid using `Function`

File: src/modules/UI/scenes/EditToken/EditTokenConnector.js [hunk 5]
```diff
@@ -2,23 +2,28 @@
...
 import {connect} from 'react-redux'
 import * as EDIT_TOKEN_ACTIONS from './action.js'
+// eslint-disable-next-line no-duplicate-imports
 import EditToken from './EditToken.ui'
+// eslint-disable-next-line no-duplicate-imports
+import type {
+  EditTokenStateProps,
+  EditTokenDispatchProps,
+  EditTokenOwnProps
+} from './EditToken.ui.js'
```
Comment (wolverineks) [inline src/modules/UI/scenes/EditToken/EditTokenConnector.js]: Please combine this into a single import statement

File: src/modules/UI/scenes/EditToken/EditTokenConnector.js [hunk 6]
```diff
@@ -2,23 +2,28 @@
...
-  dispatch,
-  showDeleteTokenModal: () => dispatch(EDIT_TOKEN_ACTIONS.showDeleteTokenModal()),
-  hideDeleteTokenModal: () => dispatch(EDIT_TOKEN_ACTIONS.hideDeleteTokenModal()),
-  deleteCustomToken: (walletId: string, currencyCode: string) => dispatch(WALLET_ACTIONS.deleteCustomToken(walletId, currencyCode)),
-  editCustomToken: (walletId: string, currencyName: string, currencyCode: string, contractAddress: string, denomination: string, oldCurrencyCode: string) =>
+const mapDispatchToProps = (dispatch: Dispatch): EditTokenDispatchProps => ({
+  showDeleteTokenModal: (): any => dispatch(EDIT_TOKEN_ACTIONS.showDeleteTokenModal()),
+  hideDeleteTokenModal: (): any => dispatch(EDIT_TOKEN_ACTIONS.hideDeleteTokenModal()),
+  deleteCustomToken: (walletId: string, currencyCode: string): any => dispatch(WALLET_ACTIONS.deleteCustomToken(walletId, currencyCode)),
+  editCustomToken: (walletId: string, currencyName: string, currencyCode: string, contractAddress: string, denomination: string, oldCurrencyCode: string): any =>
```
Comment (wolverineks) [inline src/modules/UI/scenes/EditToken/EditTokenConnector.js]: Please avoid using `any`

File: src/modules/UI/scenes/EditToken/EditTokenConnector.js [hunk 7]
```diff
@@ -2,23 +2,28 @@
...
-const mapDispatchToProps = (dispatch: Dispatch) => ({
-  dispatch,
-  showDeleteTokenModal: () => dispatch(EDIT_TOKEN_ACTIONS.showDeleteTokenModal()),
-  hideDeleteTokenModal: () => dispatch(EDIT_TOKEN_ACTIONS.hideDeleteTokenModal()),
-  deleteCustomToken: (walletId: string, currencyCode: string) => dispatch(WALLET_ACTIONS.deleteCustomToken(walletId, currencyCode)),
-  editCustomToken: (walletId: string, currencyName: string, currencyCode: string, contractAddress: string, denomination: string, oldCurrencyCode: string) =>
+const mapDispatchToProps = (dispatch: Dispatch): EditTokenDispatchProps => ({
+  showDeleteTokenModal: (): any => dispatch(EDIT_TOKEN_ACTIONS.showDeleteTokenModal()),
+  hideDeleteTokenModal: (): any => dispatch(EDIT_TOKEN_ACTIONS.hideDeleteTokenModal()),
+  deleteCustomToken: (walletId: string, currencyCode: string): any => dispatch(WALLET_ACTIONS.deleteCustomToken(walletId, currencyCode)),
```
Comment (wolverineks) [inline src/modules/UI/scenes/EditToken/EditTokenConnector.js]: Please avoid using `any`

File: src/modules/UI/scenes/EditToken/EditTokenConnector.js [hunk 8]
```diff
@@ -2,23 +2,28 @@
...
 })
-const mapDispatchToProps = (dispatch: Dispatch) => ({
-  dispatch,
-  showDeleteTokenModal: () => dispatch(EDIT_TOKEN_ACTIONS.showDeleteTokenModal()),
-  hideDeleteTokenModal: () => dispatch(EDIT_TOKEN_ACTIONS.hideDeleteTokenModal()),
-  deleteCustomToken: (walletId: string, currencyCode: string) => dispatch(WALLET_ACTIONS.deleteCustomToken(walletId, currencyCode)),
-  editCustomToken: (walletId: string, currencyName: string, currencyCode: string, contractAddress: string, denomination: string, oldCurrencyCode: string) =>
+const mapDispatchToProps = (dispatch: Dispatch): EditTokenDispatchProps => ({
+  showDeleteTokenModal: (): any => dispatch(EDIT_TOKEN_ACTIONS.showDeleteTokenModal()),
+  hideDeleteTokenModal: (): any => dispatch(EDIT_TOKEN_ACTIONS.hideDeleteTokenModal()),
```
Comment (wolverineks) [inline src/modules/UI/scenes/EditToken/EditTokenConnector.js]: Please avoid using `any`

File: src/modules/UI/scenes/EditToken/EditTokenConnector.js [hunk 9]
```diff
@@ -2,23 +2,28 @@
...
   editCustomTokenProcessing: state.ui.scenes.editToken.editCustomTokenProcessing
 })
-const mapDispatchToProps = (dispatch: Dispatch) => ({
-  dispatch,
-  showDeleteTokenModal: () => dispatch(EDIT_TOKEN_ACTIONS.showDeleteTokenModal()),
-  hideDeleteTokenModal: () => dispatch(EDIT_TOKEN_ACTIONS.hideDeleteTokenModal()),
-  deleteCustomToken: (walletId: string, currencyCode: string) => dispatch(WALLET_ACTIONS.deleteCustomToken(walletId, currencyCode)),
-  editCustomToken: (walletId: string, currencyName: string, currencyCode: string, contractAddress: string, denomination: string, oldCurrencyCode: string) =>
+const mapDispatchToProps = (dispatch: Dispatch): EditTokenDispatchProps => ({
+  showDeleteTokenModal: (): any => dispatch(EDIT_TOKEN_ACTIONS.showDeleteTokenModal()),
```
Comment (wolverineks) [inline src/modules/UI/scenes/EditToken/EditTokenConnector.js]: Please avoid using `any`

File: src/modules/UI/scenes/EditToken/components/DeleteTokenButtons.ui.js [hunk 10]
```diff
@@ -9,13 +9,13 @@ import s from '../../../../../locales/strings.js'
 import styles from '../style.js'
 import {PrimaryButton, SecondaryButton} from '../../../components/Buttons'
 
-export type Props = {
+export type DeleteTokenButtonsProps = {
   onPressCancel: Function,
   onPressDelete: Function,
```
Comment (wolverineks) [inline src/modules/UI/scenes/EditToken/components/DeleteTokenButtons.ui.js]: Please avoid using `Function`

File: src/modules/UI/scenes/EditToken/components/DeleteTokenButtons.ui.js [hunk 11]
```diff
@@ -9,13 +9,13 @@ import s from '../../../../../locales/strings.js'
 import styles from '../style.js'
 import {PrimaryButton, SecondaryButton} from '../../../components/Buttons'
 
-export type Props = {
+export type DeleteTokenButtonsProps = {
   onPressCancel: Function,
```
Comment (wolverineks) [inline src/modules/UI/scenes/EditToken/components/DeleteTokenButtons.ui.js]: Please avoid using `Function`

File: src/modules/UI/scenes/ManageTokens/ManageTokensConnector.js [hunk 12]
```diff
@@ -1,30 +1,24 @@
...
-
+// eslint-disable-next-line no-duplicate-imports
 import ManageTokens from './ManageTokens.ui.js'
-
+// eslint-disable-next-line no-duplicate-imports
+import type {
+  ManageTokensOwnProps,
+  ManageTokensDispatchProps,
+  ManageTokensStateProps
+} from './ManageTokens.ui.js'
```
Comment (wolverineks) [inline src/modules/UI/scenes/ManageTokens/ManageTokensConnector.js]: Please combine this into a single import statement

File: src/modules/UI/scenes/AddToken/AddToken.ui.js [hunk 13]
```diff
@@ -23,11 +23,24 @@ import {
...
-export type DispatchProps = {
+export type AddTokenOwnProps = {
+  walletId: string,
+  addTokenPending: Function,
+  addNewToken: Function,
+  currentCustomTokens: Array<CustomTokenInfo>,
+  wallet: GuiWallet
+}
+
+export type AddTokenDispatchProps = {
```
[the remark this answers is not in the document]
Comment (kylanhurt) [reply src/modules/UI/scenes/AddToken/AddToken.ui.js]: Done

**Pull request merged**